sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_client.hpp

```diff
@@ -242,6 +242,23 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
         const GetSharePermissionOptions& options = GetSharePermissionOptions(),
         const Azure::Core::Context& context = Azure::Core::Context()) const;
 
+    /**
+     * @brief Gets the permission of the share using the specific key.
+     * @param permissionKey The permission key of a permission.
+     * @param filePermissionFormat Optional. Available for version 2024-11-04 and later. Specifies
+     * the format in which the permission is returned.If filePermissionKeyFormat is unspecified or
+     * explicityly set to SSDL format, the permission will be
+     * returned in SSDL format.
+     * @param options Optional parameters to get share's permission.
+     * @param context Context for cancelling long running operations.
+     * @return Azure::Response<std::string> containing the permission string with specified key.
+     */
+    Azure::Response<Models::ShareFilePermission> GetPermission(
```

Comment:
can we just use Response<string> as return type?

Reply:
You mean wrapper the return value with a response type such as `GetPermissionResponse` or
we just use Response<std::string> which like previous one and don't overload this method? In this impl, user have to get the FilePermissionFormat from RawResponse.